internal/services/worker.go

```diff
@@ -27,31 +67,321 @@ func LaunchWorker(ctx context.Context, wg *sync.WaitGroup, taskService *Task) {
-		tasks, err := taskService.GetSchedulableTasks(ctx, taskLimit)
+		tasks, err := w.taskService.GetSchedulableTasks(ctx, taskLimit)
 		if err != nil {
 			logger.Errorf("Worker error fetching tasks: %v", err)
 			// Wait before retrying to avoid spamming logs on persistent DB errors
-			t.Reset(backoff)
+			t.Reset(w.backoff)
 			continue
 		}
 
 		if len(tasks) == 0 {
 			logger.Debug("Worker: No tasks to process")
 			// Wait before retrying to give time for tasks to be created
-			t.Reset(backoff)
+			t.Reset(w.backoff)
 			continue
 		}
-		// Log fetched tasks (just IDs for brevity)
-		taskIDs := make([]uint, len(tasks))
-		for i, task := range tasks {
-			taskIDs[i] = task.ID
+
+		for _, task := range tasks {
+			// Check if the context has been cancelled to avoid processing tasks after shutdown
+			select {
+			case <-ctx.Done():
+				logger.Info("Worker received shutdown signal, stopping...")
+				return
+			default:
+			}
+			switch task.Action {
+			case models.TaskActionCreateInstances:
+				err := w.processCreateInstanceTask(ctx, &task)
+				if err != nil {
+					logMsg := fmt.Sprintf("❌ Failed to process create instance task %d: %v", task.ID, err)
+					logger.Error(logMsg)
+					task.Logs += fmt.Sprintf("\n%s", logMsg)
+					err = w.taskService.UpdateFailed(ctx, &task, err.Error(), logMsg)
+					if err != nil {
+						logger.Errorf("Worker: Failed to update task: %v", err)
+					}
+					// No time reset needed here as we are just continuing through the tasks
+					continue
+				}
+				err = w.taskService.UpdateStatus(ctx, task.OwnerID, task.ID, models.TaskStatusCompleted)
+				if err != nil {
+					logger.Errorf("Worker: Failed to update task status: %v", err)
+				}
+			case models.TaskActionTerminateInstances:
+				err := w.processTerminateInstanceTask(ctx, &task)
+				if err != nil {
+					logMsg := fmt.Sprintf("❌ Failed to process terminate instance task %d: %v", task.ID, err)
+					logger.Error(logMsg)
+					task.Logs += fmt.Sprintf("\n%s", logMsg)
+					err = w.taskService.UpdateFailed(ctx, &task, err.Error(), logMsg)
+					if err != nil {
+						logger.Errorf("Worker: Failed to update task: %v", err)
+					}
+					// No time reset needed here as we are just continuing through the tasks
+					continue
+				}
+				err = w.taskService.UpdateStatus(ctx, task.OwnerID, task.ID, models.TaskStatusCompleted)
+				if err != nil {
+					logger.Errorf("Worker: Failed to update task status: %v", err)
+				}
+			default:
+				logger.Errorf("Worker: Unknown task action %s for task %d", task.Action, task.ID)
+			}
 		}
-		logger.Infof("Worker fetched %d tasks: %v", len(tasks), taskIDs)
+	}
+}
 
-		// TODO: Implement actual task processing logic here
-		// For now, we just log and discard them.
+// processCreateInstanceTask processes a create instance task. It will handle the instance creation, provisioning, and status updates for the instance.
+func (w *Worker) processCreateInstanceTask(ctx context.Context, task *models.Task) error {
```

Comment:
The intent of `processCreateInstanceTask` is the following:
1. Marks tast as running
2. Unmarshalls the instance request from the task payload
3. Determines what stage of creation the instance is in based on the status
4. Attempts to create the instance
5. Updates the instance status during each stage to help protect against shutdowns